src/models/IDPDiscovery.js

```diff
@@ -71,7 +71,7 @@ function (Okta, PrimaryAuthModel, CookieUtil, Enums) {
                   redirectToIdp: function (redirectUrl) {
                     if(res.links && res.links[0] && res.links[0].href) {
                       var queryParams = {fromURI: redirectUrl};
-                      if (res.links[0].properties && res.links[0].properties['okta:idp:type'] === 'SAML2') {
+                      if (res.links[0].properties) {
```

Comment:
- curious when `.properties` would be null?
- and when `.properties` exists, is it guaranteed that `.properties['okta:idp:type']` exists as well?

Reply:
I don't think it would ever be null. These properties include the idp type (okta, saml2, facebook, google, etc) and the idp id (for those idps that have an ID). Maybe it wouldn't hurt to just always do `queryParams['login_hint'] = username`